eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Ledgers.hs

```diff
@@ -128,24 +122,19 @@ ledgersTransition ::
   forall era.
   ( Embed (Core.EraRule "LEDGER" era) (LEDGERS era),
     Environment (Core.EraRule "LEDGER" era) ~ LedgerEnv era,
-    State (Core.EraRule "LEDGER" era) ~ (UTxOState era, DPState (Crypto era)),
+    State (Core.EraRule "LEDGER" era) ~ LedgerState era,
     Signal (Core.EraRule "LEDGER" era) ~ Core.Tx era
   ) =>
   TransitionRule (LEDGERS era)
 ledgersTransition = do
   TRC (LedgersEnv slot pp account, ls, txwits) <- judgmentContext
-  let (u, dp) = (_utxoState ls, _delegationState ls)
-  (u'', dp'') <-
-    foldM
-      ( \(u', dp') (ix, tx) ->
-          trans @(Core.EraRule "LEDGER" era) $
-            TRC (LedgerEnv slot ix pp account, (u', dp'), tx)
-      )
-      (u, dp)
-      $ zip [minBound ..] $
-        toList txwits
-
-  pure $ LedgerState u'' dp''
+  foldM
+    ( \ !ls' (ix, tx) ->
+        trans @(Core.EraRule "LEDGER" era) $
+          TRC (LedgerEnv slot ix pp account, ls', tx)
+    )
+    ls
+    $ zip [minBound ..] $ toList txwits
```

Comment:
some times it is better to use a named function as it brings the other arguments of foldM closer to the call

```
foldM accum ls (zip [minBound ..] $ toList txwits) where
   accum !ls' (ix,tx) = trans @(Core.EraRule "LEDGER" era) $ TRC (LedgerEnv slot ix pp account, ls', tx)
```

Reply:
This is really a matter of preference. I personally don't mind lambdas at all. In any case it was a lambda before, so let's keep it as such. There are enough changes for one PR already.

> It is a brave man that touches 70 files to get rid of (,) and _

FYI, this is one of the places where a memory leak is possible because `(,)` is used instead of a type with strict elements like `LedgerState`.  Note that despite that `foldM` is strict and accumulator is pattern matched on, it is only strict to whnf, so `u'` and `dp'` in the accumulator might not get evaluated. That is one of the main reasons why I am willing to put the effort into replacing tuples with strict types. With regards to `_`, their usage in our codebase is just plain wrong. It normally indicates to the reader that binding or function is unused, which is furthers from the truth in our codebase. Also, I like compiler telling me if I forgot some unused bindings, `_` masks that.